src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,623 @@
+                if (result != null)
+                {
+                    return result;
+                }
+
+                result = TryTransformDirective(lineStr, checkPrefixes, SyntaxDirectiveFullLineNext, "-NEXT");
+                if (result != null)
+                {
+                    return result;
+                }
+
+                return lineStr;
+            }
+        }
+
+        static string TransformMethod(MethodDeclarationSyntax methodDecl, string[] checkPrefixes)
+        {
+            return String.Join(Environment.NewLine, methodDecl.GetText().Lines.Select(x => TransformLine(x, checkPrefixes)));
+        }
+
+        static int GetMethodStartingLineNumber(MethodDeclarationSyntax methodDecl)
+        {
+            var leadingTrivia = methodDecl.GetLeadingTrivia();
+            if (leadingTrivia.Count == 0)
+            {
+                return methodDecl.GetLocation().GetLineSpan().StartLinePosition.Line;
+            }
+            else
+            {
+                return leadingTrivia[0].GetLocation().GetLineSpan().StartLinePosition.Line;
+            }
+        }
+
+        static string PreProcessMethod(MethodDeclarationInfo methodDeclInfo, string[] checkPrefixes)
+        {
+            var methodDecl = methodDeclInfo.Syntax;
+            var methodName = methodDeclInfo.Name;
+
+            // Create anchors from the first prefix.
+            var startAnchorText = $"// {checkPrefixes[0]}-LABEL: {methodName}(";
+            var endAnchorText = $"// {checkPrefixes[0]}: {methodName}(";
+
+            // Create temp source file based on the source text of the method.
+            // Newlines are added to pad the text so FileCheck's error messages will correspond
+            // to the correct line and column of the original source file.
+            // This is not perfect but will work for most cases.
+            var lineNumber = GetMethodStartingLineNumber(methodDecl);
+            var tmpSrc = new StringBuilder();
+            for (var i = 1; i < lineNumber; i++)
+            {
+                tmpSrc.AppendLine(String.Empty);
+            }
+            tmpSrc.AppendLine(startAnchorText);
+            tmpSrc.AppendLine(TransformMethod(methodDecl, checkPrefixes));
+            tmpSrc.AppendLine(endAnchorText);
+
+            return tmpSrc.ToString();
+        }
+
+
+        static async Task<FileCheckResult> RunSuperFileCheckAsync(MethodDeclarationInfo methodDeclInfo, string[] args, string[] checkPrefixes, string tmpFilePath)
+        {
+            File.WriteAllText(tmpFilePath, PreProcessMethod(methodDeclInfo, checkPrefixes));
+
+            try
+            {
+                args[0] = tmpFilePath;
+                return await RunLLVMFileCheckAsync(args);
+            }
+            finally
+            {
+                try { File.Delete(tmpFilePath); } catch { }
+            }
+        }
+
+        static bool IsArgumentCSharp(string arg)
```

Comment:
It's probably worth using a command line library, even for this simple command line.

Reply:
Yea, I think we should too. I don't know if we can spend much more time on making this perfect using command line though.

Reply:
I'm ok with this, though I have a question that I'll put over on the real PR.